### ISSUE TITLE
OCamlType for a type with Type Parameters that get passed is broken

### ISSUE THREAD

**Issue by mchaver** [plow-technologies/ocaml-export#14]
```haskell
data TwoTypeParameters a b =
  TwoTypeParameters
    { ttpId :: Int
    , ttpFirst :: a
    , ttpSecond :: b
    } deriving (Eq,Show,Generic,OCamlType)

λ> toOCamlType (Proxy :: Proxy (TwoTypeParameters TypeParameterRef0 TypeParameterRef1))
OCamlDatatype 
  (HaskellTypeMetaData "TwoTypeParameters" "Ghci2" "interactive") 
  "TwoTypeParameters" 
  (OCamlValueConstructor 
    (RecordConstructor 
      "TwoTypeParameters" 
       (Values 
         (OCamlField "ttpId" (OCamlPrimitiveRef OInt)) 
         (Values 
           (OCamlField "ttpFirst" (OCamlTypeParameterRef "a0")) 
           (OCamlField "ttpSecond" (OCamlTypeParameterRef "a1"))))))
```

```haskell
data Wrapper a b =
  Wrapper
    { wpa :: Int
    , wpb :: TwoTypeParameters a b
    } deriving (Eq,Show,Generic,OCamlType)

λ> toOCamlType (Proxy :: Proxy (Wrapper TypeParameterRef0 TypeParameterRef1))
OCamlDatatype 
  (HaskellTypeMetaData "Wrapper" "Ghci4" "interactive") 
  "Wrapper"
  (OCamlValueConstructor 
    (RecordConstructor "Wrapper" 
      (Values 
        (OCamlField "wpa" (OCamlPrimitiveRef OInt)) 
        (OCamlField "wpb" 
          (OCamlRef 
            (HaskellTypeMetaData "TwoTypeParameters" "Ghci2" "interactive") 
            "TwoTypeParameters")))))
```

TypeParameterRef0 and TypeParameterRef1 do not show up in Wrapper.

**Comment by mchaver** [comment]
```haskell
{-# LANGUAGE DeriveGeneric #-}
{-# OPTIONS_GHC -ddump-deriv #-}

module Generic where

import GHC.Generics

data Tree a
  = Leaf a
  | Node (Tree a) (Tree a)
  deriving Generic

data Wrapper =
  Wrapper
    { wpa :: Int
    , wpb :: Tree String
    } deriving Generic
```

```
GHC.Generics representation types:
  type GHC.Generics.Rep (Generic.Tree a_a5Cj) = GHC.Generics.D1
                                                  ('GHC.Generics.MetaData
                                                     "Tree" "Generic" "main" 'GHC.Types.False)
                                                  (GHC.Generics.C1
                                                     ('GHC.Generics.MetaCons
                                                        "Leaf"
                                                        'GHC.Generics.PrefixI
                                                        'GHC.Types.False)
                                                     (GHC.Generics.S1
                                                        ('GHC.Generics.MetaSel
                                                           'GHC.Base.Nothing
                                                           'GHC.Generics.NoSourceUnpackedness
                                                           'GHC.Generics.NoSourceStrictness
                                                           'GHC.Generics.DecidedLazy)
                                                        (GHC.Generics.Rec0 a_a5Cj))
                                                   GHC.Generics.:+: GHC.Generics.C1
                                                                      ('GHC.Generics.MetaCons
                                                                         "Node"
                                                                         'GHC.Generics.PrefixI
                                                                         'GHC.Types.False)
                                                                      (GHC.Generics.S1
                                                                         ('GHC.Generics.MetaSel
                                                                            'GHC.Base.Nothing
                                                                            'GHC.Generics.NoSourceUnpackedness
                                                                            'GHC.Generics.NoSourceStrictness
                                                                            'GHC.Generics.DecidedLazy)
                                                                         (GHC.Generics.Rec0
                                                                            (Generic.Tree a_a5Cj))
                                                                       GHC.Generics.:*: GHC.Generics.S1
                                                                                          ('GHC.Generics.MetaSel
                                                                                             'GHC.Base.Nothing
                                                                                             'GHC.Generics.NoSourceUnpackedness
                                                                                             'GHC.Generics.NoSourceStrictness
                                                                                             'GHC.Generics.DecidedLazy)
                                                                                          (GHC.Generics.Rec0
                                                                                             (Generic.Tree
                                                                                                a_a5Cj))))
  type GHC.Generics.Rep Generic.Wrapper = GHC.Generics.D1
                                            ('GHC.Generics.MetaData
                                               "Wrapper" "Generic" "main" 'GHC.Types.False)
                                            (GHC.Generics.C1
                                               ('GHC.Generics.MetaCons
                                                  "Wrapper" 'GHC.Generics.PrefixI 'GHC.Types.True)
                                               (GHC.Generics.S1
                                                  ('GHC.Generics.MetaSel
                                                     ('GHC.Base.Just "wpa")
                                                     'GHC.Generics.NoSourceUnpackedness
                                                     'GHC.Generics.NoSourceStrictness
                                                     'GHC.Generics.DecidedLazy)
                                                  (GHC.Generics.Rec0 GHC.Types.Int)
                                                GHC.Generics.:*: GHC.Generics.S1
                                                                   ('GHC.Generics.MetaSel
                                                                      ('GHC.Base.Just "wpb")
                                                                      'GHC.Generics.NoSourceUnpackedness
                                                                      'GHC.Generics.NoSourceStrictness
                                                                      'GHC.Generics.DecidedLazy)
                                                                   (GHC.Generics.Rec0
                                                                      (Generic.Tree
                                                                         GHC.Base.String))))

```

**Comment by mchaver** [comment]
Resolved in version 0.6.0.0 by this PR
https://github.com/plow-technologies/ocaml-export/pull/15
